pkg/controller/kappnav/kappnav_controller.go

```diff
@@ -450,21 +581,83 @@ func (r *ReconcileKappnav) Reconcile(request reconcile.Request) (reconcile.Resul
 			Namespace: instance.GetNamespace(),
 		},
 	}
-	err = r.CreateOrUpdate(controllerDeployment, instance, func() error {
+	if logger.IsEnabled(kappnavutils.LogTypeInfo) {
+		logger.Log(kappnavutils.CallerName(), kappnavutils.LogTypeInfo, "Create or update controller deployment"+otherLogData, logName)
+	}
+	err = r.CreateOrUpdate(logger, controllerDeployment, instance, func() error {
 		pts := &controllerDeployment.Spec.Template
 		kappnavutils.CustomizeDeployment(controllerDeployment, instance)
 		kappnavutils.CustomizePodSpec(pts, &controllerDeployment.ObjectMeta,
 			kappnavutils.CreateControllerDeploymentContainers(pts.Spec.Containers, instance), nil, instance)
 		return nil
 	})
 	if err != nil {
-		reqLogger.Error(err, "Failed to reconcile the Controller Deployment")
-		return r.ManageError(err, kappnavv1.StatusConditionTypeReconciled, instance)
+		if logger.IsEnabled(kappnavutils.LogTypeError) {
+			logger.Log(kappnavutils.CallerName(), kappnavutils.LogTypeError, fmt.Sprintf("Failed to reconcile the Controller Deployment"+otherLogData+", Error: %s", err), logName)
+		}
+		return r.ManageError(logger, err, kappnavv1.StatusConditionTypeReconciled, instance)
+	}
+
+	// Apply defaults to the KindActionMapping (kam) instance
+	default_kam := &kamv1.KindActionMapping{}
```

Comment:
I would suggest applying the default KAM before the creation of the Deployments since they depend on this resource existing.

Reply:
ok